backend/hitas/views/sale.py

```diff
@@ -0,0 +1,93 @@
+from typing import Any
+from uuid import UUID
+
+from django.core.exceptions import ValidationError
+from django.db.models import Prefetch
+
+from hitas.exceptions import HitasModelNotFound
+from hitas.models import Apartment, ApartmentSale
+from hitas.models.ownership import Ownership, OwnershipLike, check_ownership_percentages
+from hitas.views.ownership import OwnershipSerializer
+from hitas.views.utils import HitasModelSerializer, HitasModelViewSet
+
+
+class ApartmentSaleSerializer(HitasModelSerializer):
+
+    ownerships = OwnershipSerializer(many=True)
+
+    def to_internal_value(self, data: dict[str, Any]) -> dict[str, Any]:
+        # Add empty list here for os that ownerships is can be left out when updating.
+        data.setdefault("ownerships", [])
+        return super().to_internal_value(data)
+
+    def validate_ownerships(self, ownerships: list[OwnershipLike]) -> list[OwnershipLike]:
+        if self.context["view"].action != "create":
+            if ownerships:
+                raise ValidationError("Can't update ownerships.")
+            return ownerships
+
+        if not ownerships:
+            raise ValidationError("Sale must have ownerships.")
+
+        check_ownership_percentages(ownerships)
+        return ownerships
+
+    def get_apartment_id(self) -> int:
+        try:
+            apartment_uuid = UUID(hex=self.context["view"].kwargs.get("apartment_uuid"))
+            apartment_id = Apartment.objects.only("id").get(uuid=apartment_uuid).id
+        except (Apartment.DoesNotExist, ValueError) as error:
+            raise HitasModelNotFound(model=Apartment) from error
+        return apartment_id
```

Comment:
IIRC we have some reusable function for this, if not maybe one should be created (At least noted down somewhere so we remember to improve this) 😄

Reply:
Refactored reusable functions "lookup_model_id_by_uuid" and "lookup_id_to_uuid" from under "HitasModelMixin" so that they can be more easily used in serializers.